protocol/message.go

```diff
@@ -120,3 +131,110 @@ func NewMonitoringProof(ap []*m.AuthenticationPath,
 		},
 	}, e
 }
+
+// Verify implementation
```

Comment:
Or we should have a `verification.go` and put all these functions there, except `Verify` interface. What do you think?